common_code/inspiral/CE_orbit.f90

```diff
@@ -76,66 +91,163 @@ subroutine CE_orbit_adjust(id, ierr)
-
-         ! Given the final energy, E_final, determine the resulting k that solves the equation
-         top = 2.0 * E_final * R_int + standard_cgrav * CE_companion_mass * Msun * M_int
-         bottom = 2.0 * E_final * R_slope + standard_cgrav * CE_companion_mass * Msun * M_slope
-         k_final = -top / bottom
-
-         ! Now use the interpolations and the derived k_final, determine the resulting separation and enclosed mass
-         R_final = R_slope * k_final + R_int
-         M_final = M_slope * k_final + M_int
-
-         s% xtra(2) = R_final/Rsun
-         !Saving as s% xtra(9) the enclosed mass so that we output it in the history data
-         s% xtra(9) = M_final/Msun
-
-         ! Calculate the angular momentum lost to the star's envelope
-         J_tmp = (CE_companion_mass * Msun)**2 * M_final**2 / (CE_companion_mass * Msun + M_final)
-         J_final = sqrt(standard_cgrav * J_tmp * R_final)
-
-         !The angular momentum that is lost from the orbit of the companion
-         ! is added to the envelope of the donor.
-         orbital_ang_mom_lost = J_final - J_init
-         !We save in s% xtra(6) the total torque that will be applied to the Envelope
-         s% xtra(6) = max(0.0d0, -orbital_ang_mom_lost/s% dt) 
-
-         ! Keep track of orbital energy and angular momentum
-         s% xtra(8) = E_final
-         s% xtra(10) = J_final
+         if (s% x_logical_ctrl(7)) then
+            ! Complete integration of the eqs. of motion
+
+            if (s% model_number .eq. 1) then
+               ! Determine the initial conditions of the velocity
+               if (s% x_logical_ctrl(8)) then
+                  s% xtra(25) = s% x_ctrl(2) * s% r(1) / Rsun
+                  ! Circular orbit
+                  s% xtra(25) = pi    ! nu (rad)
+                  s% xtra(26) = 0.0d0 ! dr/dt
+
+                  ! Finding the enclosed mass by the position of the companion 
+                  call FindApproximateValueIndex(s% r , s% x_ctrl(2) * s% r(1) , index)
+
+                  ! Obtain the circular Keplerian velocity for the companion
+                  s% xtra(27) =   2.d0*pi/AtoP(M_encl, s% xtra(4)*Msun , CE_companion_position * Rsun) * secyer ! Compute Keplerian velocity (rad/yr)
+
+                  init_cond = [ CE_companion_position ,s% xtra(26), s% xtra(25) , s% xtra(27)]
+
+                  ! Get the components of the acceleration
+                  call read_mesa_id(id,ierr)
+                  call kepler_polar(prop,s% star_age, init_cond, f)
+                  s% xtra(28) = f(2) ! (Rsun / yr^2)
+                  s% xtra(29) = f(4) ! (rad / yr^2)
+                  
+               else
+                  ! Velocity's components defined by de user
+                  s% xtra(25) = s% x_ctrl(2) * s% r(1) / Rsun
+                  s% xtra(25) = pi            ! nu (rad)
+                  s% xtra(26) = s% x_ctrl(20) ! dr/dt (Rsun/yr)
+                  s% xtra(27) = s% x_ctrl(21) ! dnu/dt (rad/yr)
```

Comment:
Beside providing a 2D velocity, should we allow to provide an eccentricity instead?
Should we allow any velocity, e.g. angular velocity (0 would be no orbit but a direct hit towards center; negative and positive should be symmetric)?